### PR TITLE
Validate operating profit (or loss) total

### PR DESCRIPTION
**Pull request by faisal-amin82**
- Added validation for operating total and test cases
 
Resolves: SFA-1936

### CODE REVIEW

**Comment by Jamie87** [comment]
Hi Faisal, I cant see any unit tests for the operating p&l validator. It needs to have seperate unit tests for the new section

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/ProfitAndLossValidator.java [hunk 1]
```diff
@@ -24,60 +23,108 @@
...
+                .orElse(0L);
+    }
+
+    public void validateOperatingTotal(ProfitAndLoss profitAndLoss,  Errors errors) {
+
+        Long administrativeExpenses = getAdministrativeExpenses(profitAndLoss.getOperatingProfitOrLoss());
+
+        Long distributionCosts = getDistributionCosts(profitAndLoss.getOperatingProfitOrLoss());
+
+        Long total = getOperatingTotal(profitAndLoss.getOperatingProfitOrLoss()) ;
```
[the remark this answers is not in the document]
Comment (faisal-amin82) [reply src/main/java/uk/gov/companieshouse/api/accounts/validation/ProfitAndLossValidator.java]: renamed as suggested

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/ProfitAndLossValidator.java [hunk 2]
```diff
@@ -24,60 +23,108 @@
...
+
+        Long otherOperatingIncome = getOtherOperatingIncome(profitAndLoss.getOperatingProfitOrLoss());
+
+        Long grossProfitOrLoss = profitAndLoss.getGrossProfitOrLoss().getGrossTotal();
+
+        Long operatingTotal =  grossProfitOrLoss - distributionCosts - administrativeExpenses + otherOperatingIncome;
+
+        if(!operatingTotal.equals(total)) {
```
[the remark this answers is not in the document]
Comment (faisal-amin82) [reply src/main/java/uk/gov/companieshouse/api/accounts/validation/ProfitAndLossValidator.java]: same as above

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/ProfitAndLossValidator.java [hunk 3]
```diff
@@ -24,60 +23,108 @@
...
+
+        Long distributionCosts = getDistributionCosts(profitAndLoss.getOperatingProfitOrLoss());
+
+        Long total = getOperatingTotal(profitAndLoss.getOperatingProfitOrLoss()) ;
+
+        Long otherOperatingIncome = getOtherOperatingIncome(profitAndLoss.getOperatingProfitOrLoss());
+
+        Long grossProfitOrLoss = profitAndLoss.getGrossProfitOrLoss().getGrossTotal();
+
+        Long operatingTotal =  grossProfitOrLoss - distributionCosts - administrativeExpenses + otherOperatingIncome;
```
[the remark this answers is not in the document]
Comment (faisal-amin82) [reply src/main/java/uk/gov/companieshouse/api/accounts/validation/ProfitAndLossValidator.java]: renamed as suggested

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/ProfitAndLossValidator.java [hunk 4]
```diff
@@ -24,60 +23,108 @@
...
+
+        Long administrativeExpenses = getAdministrativeExpenses(profitAndLoss.getOperatingProfitOrLoss());
+
+        Long distributionCosts = getDistributionCosts(profitAndLoss.getOperatingProfitOrLoss());
+
+        Long total = getOperatingTotal(profitAndLoss.getOperatingProfitOrLoss()) ;
+
+        Long otherOperatingIncome = getOtherOperatingIncome(profitAndLoss.getOperatingProfitOrLoss());
+
+        Long grossProfitOrLoss = profitAndLoss.getGrossProfitOrLoss().getGrossTotal();
```
[the remark this answers is not in the document]
Comment (faisal-amin82) [reply src/main/java/uk/gov/companieshouse/api/accounts/validation/ProfitAndLossValidator.java]: Changed the code as suggested in the comments. We now validate the Gross Total before using it

**Pull request merged**